GLOSSARY.md

```diff
@@ -1,85 +1,113 @@
 # PyTorch Glossary
 
+Several terms in this glossary are used frequently throughout PyTorch documentation and warrant basic definitions.
+
 <!-- toc -->
 
-- [Operation and Kernel](#operation-and-kernel)
-  - [ATen](#aten)
-  - [Operation](#operation)
-  - [Native Operation](#native-operation)
-  - [Custom Operation](#custom-operation)
-  - [Kernel](#kernel)
-  - [Compound Operation](#compound-operation)
-  - [Composite Operation](#composite-operation)
-  - [Non-Leaf Operation](#non-leaf-operation)
-  - [Leaf Operation](#leaf-operation)
-  - [Device Kernel](#device-kernel)
-  - [Compound Kernel](#compound-kernel)
-- [JIT Compilation](#jit-compilation)
-  - [JIT](#jit)
-  - [TorchScript](#torchscript)
-  - [Tracing](#tracing)
-  - [Scripting](#scripting)
+- [PyTorch Glossary](#pytorch-glossary)
+  - [Model, Tensor, Operation and Kernel](#model-tensor-operation-and-kernel)
+    - [Model](#model)
+    - [Tensor](#tensor)
+    - [ATen](#aten)
+    - [Operation](#operation)
+    - [Native Operation](#native-operation)
+    - [Custom Operation](#custom-operation)
+    - [Kernel](#kernel)
+    - [Compound Operation](#compound-operation)
+    - [Composite Operation](#composite-operation)
+    - [Non-Leaf Operation](#non-leaf-operation)
+    - [Leaf Operation](#leaf-operation)
+    - [Device Kernel](#device-kernel)
+    - [Compound Kernel](#compound-kernel)
+  - [JIT Compilation](#jit-compilation)
+    - [JIT](#jit)
+    - [TorchScript](#torchscript)
+    - [Tracing](#tracing)
+    - [Scripting](#scripting)
 
 <!-- tocstop -->
 
-# Operation and Kernel
+## Model, Tensor, Operation and Kernel
+
+### Model
+
+An algorithm for making predictions from data. (E.g. AI Model)
+
+### Tensor
+
+(1:formal) A multilinear map, in the form of a specialized storage data structure, similar to arrays and matrices.
+(2:informal) A multi-dimensional array/matrix containing elements (data) of a single data type (int, float, ...).
+(3:PyTorch implementation) A low-level representation of the specialized storage data structure, containing pointers to storage where the data and metadata are located.
+
+- In PyTorch tensors are used to encode the data inputs & outputs of a model and the model’s parameters.
```

Comment:
I suggest that the PyTorch Tensor is like the NumPy array an supports similar operations. The PyTorch tensor also includes the device it is running on and the autograd function when present.

Reply:
> I suggest that the PyTorch Tensor is like the NumPy array an supports similar operations.

nod.. fyi:  not only is it like NumPy see https://pytorch.org/docs/stable/generated/torch.Tensor.numpy.html  

do you have a suggested change?

> The PyTorch tensor also includes the device it is running on and the autograd function when present.

I intentionally skipped the meta data content here, maybe a link to the type/struct definition.

Reply:
ok I wonder why you skipped the metadata content , my thought was that this metadata is the major difference between a numpy array and a PyTorch Tensor, and dictate where and how operations will be processed. Maybe we could mention them and link out to the details? What do you think?

Reply:
> ok I wonder why you skipped the metadata content
...
> and dictate where and how operations will be processed

 a lot there, avoiding the rabbit hole where the glossary turns into a design document.

> link out to the details

nod.. maybe a link to the type/struct definition.

> my thought was that this metadata is the major difference between a numpy array and a PyTorch Tensor

to be honest, python isn't my primary programming language.. thus, I didn't see numpy as an important thing to talk about here :-O

Already skating the line by talking about multilinear map.. presuming the skill set is a programmer familiar with maps and linear algebra.. probably should do links there to.

Reply:
thoughts?